pandas/core/ops.py

```diff
@@ -1241,23 +1241,22 @@ def _bool_method_SERIES(cls, op, special):
     Wrapper function for Series arithmetic operations, to avoid
     code duplication.
     """
+    name = _get_op_name(op, special)
 
     def na_op(x, y):
         try:
             result = op(x, y)
         except TypeError:
-            if isinstance(y, list):
-                y = construct_1d_object_array_from_listlike(y)
-
-            if isinstance(y, (np.ndarray, ABCSeries)):
-                if (is_bool_dtype(x.dtype) and is_bool_dtype(y.dtype)):
-                    result = op(x, y)  # when would this be hit?
-                else:
-                    x = _ensure_object(x)
-                    y = _ensure_object(y)
-                    result = libops.vec_binop(x, y, op)
+            assert not isinstance(y, (list, ABCSeries, pd.Index))
```

Comment:
add tuple here, *or* assert ``is_list_like`` and not ``np.ndarray``

Reply:
or maybe just remove the assertion and make the else into an ``is_scalar`` and make an else that raises an AssertionError

Reply:
As mentioned above, I’m not inclined to change any behavior until the docs/tests are fleshed out.

Reply:
then let's do that first (in this PR or a pre-cursor)